src/resource-archive/index.ts

```diff
@@ -21,6 +21,10 @@ class Watcher {
 
   private client: CDPSession;
 
+  private unfulfilledRequests: Map<string, Promise<any>>;
+
+  private unsettledResolvers: Map<string, any>;
```

Comment:
I think these names could be more descriptive. Also why is the second one `string=>any`?

I'm not sure I quite see why they need to be separate lists either?

Reply:
The first one holds `Promise`s, but the second one holds functions that resolve `Promise`s. They probably don't need to be separate lists - I'm not 100% sure why I separated them, other than I think I was experimenting with a different way of doing some of these. Honestly, I don't think that the `unsettledResolvers` really needs to exist. I'll refactor this a bit and see if I can coalesce them.